Fix cyclic import in lib/condor.py

https://github.com/fermitools/jobsub_lite/blob/f222428da0026016d8e156a640256e2f6b99eb38/lib/condor.py#L31

PR #508 introduced a cyclic import in lib/condor.py, and I've currently disabled the `cyclic-import` pylint check on the line indicated above. 

As indicated in #513, we should fix this before we release 1.6, and we should re-enable the `cyclic-import` check to make sure that it's truly resolved.

Steps to reproduce:

1.  Checkout current master (f222428da0)
2. Delete line 31 in lib/condor.py (the line reads `# pylint: disable=cyclic-import`
3. Run pre-commit on all the files:
```
$ pre-commit run --all-files
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
mypy (no tests or bin)...................................................Passed
mypy (bin)...............................................................Passed
pylint...................................................................Failed
- hook id: pylint
- exit code: 12

************* Module version
lib/version.py:1:0: R0401: Cyclic import (condor -> submit_support) (cyclic-import)
lib/version.py:1:0: R0401: Cyclic import (condor -> submit_support -> dagnabbit -> get_parser) (cyclic-import)
lib/version.py:1:0: R0401: Cyclic import (condor -> submit_support -> dagnabbit -> get_parser -> pool) (cyclic-import)

------------------------------------------------------------------
Your code has been rated at 9.98/10 (previous run: 9.77/10, +0.21)

black....................................................................Passed
```

--- a/lib/condor.py
+++ b/lib/condor.py
@@ -33,6 +33,7 @@ import fake_ifdh
 import packages
 from render_files import render_files
 from tracing import as_span
+from transfer_sandbox import transfer_sandbox
 
 PREFIX = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
@@ -434,11 +435,6 @@ def submit_dag(
         #    cmd = f"BEARER_TOKEN_FILE={os.environ['BEARER_TOKEN_FILE']} {cmd}"
 
         if vargs["outurl"]:
-            # pylint: disable=import-outside-toplevel
-            from submit_support import (
-                transfer_sandbox,
-            )
-
             transfer_sandbox(vargs["outdir"], vargs["outurl"])
 
         if vargs.get("verbose", 0) > 0:

--- a/lib/condor.py
+++ b/lib/condor.py
@@ -28,7 +28,6 @@ import classad  # type: ignore
 import htcondor  # type: ignore
 import jinja2  # type: ignore
 
-# pylint: disable=cyclic-import
 import fake_ifdh
 import packages
 from render_files import render_files

--- a/lib/fake_ifdh.py
+++ b/lib/fake_ifdh.py
@@ -162,6 +162,7 @@ def getRole_from_valid_token() -> Optional[str]:
     return None
 
 
+# pylint: disable=unused-argument
 @as_span("getRole")
 def getRole(role_override: Optional[str] = None, verbose: int = 0) -> str:
     """get current role.  Will check the following in order:

--- a/lib/submit_support.py
+++ b/lib/submit_support.py
@@ -6,9 +6,8 @@ from typing import List, Dict, Any
 
 from condor import submit, submit_dag
 from dagnabbit import parse_dagnabbit
-from fake_ifdh import mkdir_p, cp
 from render_files import render_files
-from tracing import as_span
+from transfer_sandbox import transfer_sandbox
 
 PREFIX = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
@@ -61,30 +60,6 @@ def do_dataset_defaults(varg: Dict[str, Any]) -> None:
         varg["environment"].append(f"SAM_USER={os.environ['USER']}")
     if not have_group:
         varg["environment"].append(f"SAM_GROUP={experiment}")
-
-
-@as_span("transfer_sandbox")
-def transfer_sandbox(src_dir: str, dest_url: str) -> None:
-    """Transfer files from src_dir to sandbox with fake_ifdh (gfal-copy).
-    Nothing failing here is considered fatal, since it doesn't affect the job
-    itself, just log availability.
-
-    """
-    print("Transferring files to web sandbox...")
-    try:
-        mkdir_p(dest_url)
-    except Exception as e:  # pylint: disable=broad-except
-        print(
-            f"warning: error creating sandbox, web logs will not be available for this submission: {e}"
-        )
-        return
-    for f in os.listdir(src_dir):
-        try:
-            cp(os.path.join(src_dir, f), os.path.join(dest_url, f))
-        except Exception as e:  # pylint: disable=broad-except
-            print(
-                f"warning: error copying {f} to sandbox, will not be available through web logs: {e}"
-            )
 
 
 def get_env_list(name: str) -> List[str]:

--- a/lib/transfer_sandbox.py
+++ b/lib/transfer_sandbox.py
@@ -1,0 +1,32 @@
+# pylint: disable=wrong-import-position,wrong-import-order,import-error
+import os
+import os.path
+
+from fake_ifdh import mkdir_p, cp
+from tracing import as_span
+
+PREFIX = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+@as_span("transfer_sandbox")
+def transfer_sandbox(src_dir: str, dest_url: str) -> None:
+    """Transfer files from src_dir to sandbox with fake_ifdh (gfal-copy).
+    Nothing failing here is considered fatal, since it doesn't affect the job
+    itself, just log availability.
+
+    """
+    print("Transferring files to web sandbox...")
+    try:
+        mkdir_p(dest_url)
+    except Exception as e:  # pylint: disable=broad-except
+        print(
+            f"warning: error creating sandbox, web logs will not be available for this submission: {e}"
+        )
+        return
+    for f in os.listdir(src_dir):
+        try:
+            cp(os.path.join(src_dir, f), os.path.join(dest_url, f))
+        except Exception as e:  # pylint: disable=broad-except
+            print(
+                f"warning: error copying {f} to sandbox, will not be available through web logs: {e}"
+            )

